python/tests/test_tables.py

```diff
@@ -851,6 +851,32 @@ def test_round_trip_set_columns(self):
                 assert table[j].metadata == metadata_column[j]
                 assert table[j + num_rows].metadata == metadata_column[j]
 
+    def test_set_null_metadata(self):
+        table = self.table_class()
+        table.metadata_schema = metadata.MetadataSchema(
+            {
+                "codec": "struct",
+                "title": "Example Metadata",
+                "type": ["object", "null"],
+                "properties": {
+                    "a": {"type": "number", "binaryFormat": "i"},
+                },
+                "required": ["a"],
+                "additionalProperties": False,
+            },
+        )
+        metadata_list = [{"a": 4}, None, {"a": 5}]
```

Comment:
Might as well add `{}` here too to check it doesn't become `None` with future fiddling.

Reply:
That won't validate on this schema, but I'll add another test.